components/board/digital_interrupts.go

```diff
@@ -15,6 +15,13 @@ import (
 // servo ticks.
 const ServoRollingAverageWindow = 10
 
+// Tick represents a signal received by an interrupt pin. This signal is communicated
+// via registered channel to the various drivers.
+type Tick struct {
+	High              bool
+	TimestampMicroSec uint32
```

Comment:
Discussion question: part of me says keeping this as an integer is the right choice, because it's fast and very obvious what to do with it. Part of me says it should be a `time.Time` object instead, so you can't accidentally treat this as nanoseconds or (regular) seconds. However, constructing a `time.Time` object might incur extra computation, which might slow down the thing you're trying to time. What do other folks think? I half imagine that Gautham already tried this route, and hit some subtle problem I haven't thought of.

Reply:
It make sense that we should wrap the timestamp in a time.Time struct, the extra slow down won't matter since the ground truth for what we are measuring is from the tick function.